src/client/interpreter/autoSelection/rules/currentPath.ts

```diff
@@ -17,17 +25,21 @@ export class CurrentPathInterpretersAutoSelectionRule extends BaseRuleService {
         @inject(IFileSystem) fs: IFileSystem,
         @inject(IInterpreterHelper) private readonly helper: IInterpreterHelper,
         @inject(IPersistentStateFactory) stateFactory: IPersistentStateFactory,
-        @inject(IInterpreterLocatorService)
-        @named(CURRENT_PATH_SERVICE)
-        private readonly currentPathInterpreterLocator: IInterpreterLocatorService,
+        @inject(IComponentAdapter) private readonly pyenvs: IComponentAdapter,
+        @inject(IExperimentService) private readonly experimentService: IExperimentService,
+        @inject(IServiceContainer) private readonly serviceContainer: IServiceContainer,
     ) {
         super(AutoSelectionRule.currentPath, fs, stateFactory);
     }
     protected async onAutoSelectInterpreter(
         resource: Resource,
         manager?: IInterpreterAutoSelectionService,
     ): Promise<NextAction> {
-        const interpreters = await this.currentPathInterpreterLocator.getInterpreters(resource);
+        let interpreters = (await inDiscoveryExperiment(this.experimentService))
+            ? await this.pyenvs.getInterpreters(resource, undefined, [PythonEnvSource.PathEnvVar])
```

Comment:
Why not use a more specific method, like we are in other cases?

```suggestion
            ? await this.pyenvs.getPATHInterpreters(resource)
```

Reply:
The method was added before this PR, I just used it. But thinking about it, `source` is similar to how `query` is in `iterEnvs`. We have `query.kind` using which we can filter kind, similarly we're filtering using `source`. So I think it's okay.